v2/i18n/bundle.go

```diff
@@ -26,9 +26,58 @@ type Bundle struct {
 	matcher          language.Matcher
 }
 
+// The matcher in x/text/language does not handle artificial languages,
+// see https://github.com/golang/go/issues/45749
+// This is a simplified matcher that delegates to the x/text/language matcher for
+// the harder cases.
+type matcher struct {
+	tags           []language.Tag
+	defaultMatcher language.Matcher
+}
+
+func newMatcher(tags []language.Tag) language.Matcher {
+	var hasArt bool
+	for _, tag := range tags {
+		base, _ := tag.Base()
+		hasArt = base == artTagBase
+		if hasArt {
+			break
+		}
+	}
+
+	if !hasArt {
+		return language.NewMatcher(tags)
+	}
+
+	return matcher{
```

Comment:
> Hugo creates one Localizer per language

Oh interesting. Can we limit the scope of this workaround even further? For example, the new matcher should only activate when ALL the tags are artificial?

This is fundamentally an upstream issue so I don’t want to add a lot of non-intuitive complexity to go-i18n itself.

Reply:
>the new matcher should only activate when ALL the tags are artificial?

No, the tags set here is all the tags in the bundle. There will be a mix of real and artificial languages. 

>This is fundamentally an upstream issue so I don’t want to add a lot of non-intuitive complexity to go-i18n itself.

I'm just talking for myself here, but to me this is **my issue**; I had an extremely naive test with only 1 artificial language -- so with that (my) upgrade to the v2 package of go-i18n broke many Hugo sites in the wild, which is why I have said in another thread that I'm releasing a patch for this this week, one way or another.

Reply:
OK, I thought this patch was uncontroversial -- I'll try to find another temporary workaround. Thanks.